hooks/checkout

```diff
@@ -8,8 +8,25 @@ if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" ]]; then
     exit 0
 fi
 
-WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "--- :open_file_folder: Setting up workspace"
+WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "Creating directory: $WORKSPACE"
 mkdir -p "$WORKSPACE" && cd "$WORKSPACE"
 echo "Setup completed successfully."
+
+echo "--- Checking out repository"
+echo "Setting up git and ssh..."
+# setup git to use ssh instead of https
+git config --global url."ssh://git@github.com".insteadOf "https://github.com"
+mkdir -p "$HOME/.ssh" && ssh-keyscan -t rsa github.com >> "$HOME/.ssh/known_hosts"
```

Comment:
Oh, so decided to go with a keyscan everytime. May I know the rationale behind this? (over the other option we had)

Reply:
keyscan would not fail if the public key corresponding to `github.com` ever changes(which I doubt has a very small chance of happening). Both options are equally good I think. For added security I can add a step to confirm the RSA fingerprint of the public key that we get from keyscan.

On another note, since we are not purging the `.ssh` directory or the `known_hosts` file after every job, does it make sense to append the key to the `known_hosts` file every time? Because that would just create duplicate entries.

Reply:
Updated logic to take care of both concerns